sdks/uniswapx-sdk/src/order/types.ts

```diff
@@ -114,3 +119,32 @@ export type PriorityInputJSON = Omit<
 export type PriorityOutputJSON = PriorityInputJSON & {
   recipient: string;
 };
+
+export type V3DutchInput = {
+  readonly token: string;
+  readonly startAmount: BigNumber;
+  readonly curve: V3Decay;
+  readonly maxAmount: BigNumber;
+};
+
+export type V3DutchInputJSON = Omit<V3DutchInput, "startAmount" | "curve" | "maxAmount"> & {
+  startAmount: string;
+  curve: V3Decay;
+  maxAmount: string;
+};
+
+export type V3Decay = {
+  relativeBlocks: number[];
+  relativeAmounts: BigNumber[]; //amounts plural
+};
+
+export type V3DutchOutput = {
+  readonly token: string;
+  readonly startAmount: BigNumber;
+  readonly curve: V3Decay;
+  readonly recipient: string;
+};
+
+export type V3DutchOutputJSON = Omit<V3DutchOutput, "startAmount"> & {
+  startAmount: string;
```

Comment:
don't you need to serialize the BigInt values in the decay curve too? For both input and output

Reply:
fixed, serialized them to strings for JSON